src/Forms/DiffField.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace SilverStripe\VersionedAdmin\Forms;
+
+use LogicException;
+use SilverStripe\Forms\FormField;
+use SilverStripe\Forms\HTMLReadonlyField;
+use SilverStripe\View\Parsers\Diff;
+
+class DiffField extends HTMLReadonlyField
+{
+    /**
+     * @var FormField Holds the field used as the 'To' (the newer) data for the diff.
+     */
+    protected $comparisonField;
+
+    /**
+     * @param FormField
+     * @return $this
+     */
+    public function setComparisonField($field)
+    {
+        if (!$field instanceof FormField) {
+            throw LogicException('$field is not an instance of ' . FormField::class);
+        }
+        $this->comparisonField = $field;
+        $this->setFailover($this->comparisonField);
+        return $this;
+    }
+
+    /**
+     * @return FormField
+     */
+    public function getComparisonField()
+    {
+        return $this->comparisonField;
+    }
+
+    public function Value() {
+        $oldValue = $this->getOutdatedField()->Value();
+        $newValue = $this->getComparisonField()->Value();
+        return Diff::compareHTML($oldValue, $newValue);
+    }
+
+    public function getOutdatedField() {
```

Comment:
I reckon we could probably use better names. You may not necessarily be comparing an old version against a new version, in fact it could be the opposite to that. In that case, neither field is "outdated." Maybe `getCompareFromField` and `getCompareToField` or even `getFromField` and `getToField`

Reply:
Ha, I thought long an hard about this. I had from and to and the class got a bit unmanageable (keeping track of two fields instead of one). In the end I considered @ clarkepaul's request that comparisons always be older to newer (chronologically), but you are correct in that this form field could be used elsewhere.

>`diff [-c| -e| -f| -C n][-br] file1 file2`
>
>**DESCRIPTION**
>The diff utility shall compare the contents of `file1` and `file2` and write to standard output a list of changes necessary _to convert `file1` into `file2`_. This list should be minimal. No output shall be produced if the files are identical.

I settled on 'outdated' in the manner that a `diff(1)` is always what it takes to turn `from` _into_ `to`, leading me to think in terms of `outdated` to `updated`, even if that update is a reversal... kinda like accounting (or say, `git`, you undo by re-doing in the negative, as such).

**tl;dr**
So in that regard `outdated` is 'old' not in terms of chronology, but in terms of _succession_.

Would you be happy to settle for a comment (/docblock) summarising this?

Reply:
>In the end I considered @ clarkepaul's request that comparisons always be older to newer (chronologically), but you are correct in that this form field could be used elsewhere.

Yeah, that's a UI request AFAIK. This field will probably end up in core and could be used for other things (like advanced workflow)

Reply:
Yeah, doc blocks would be good